src/net_processing.cpp

```diff
@@ -911,12 +911,10 @@ bool static AlreadyHave(const CInv& inv) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
                 recentRejects->reset();
             }
 
-            // Use pcoinsTip->HaveCoinsInCache as a quick approximation to exclude
-            // requesting or processing some txs which have already been included in a block
             return recentRejects->contains(inv.hash) ||
                    mempool.exists(inv.hash) ||
                    mapOrphanTransactions.count(inv.hash) ||
-                   pcoinsTip->HaveCoinsInCache(inv.hash);
+                   pcoinsTip->HaveCoinsInCache(COutPoint(inv.hash, 0)); // Best effort: only try output 0
```

Comment:
It may be advisable to also check 1.  The reason is that recently confirmed transactions will get requested again, and we're currently protected by this check. But if the transaction chained off output 0 it won't be in the cache anymore. Also testing 1 would be a lot less often that we would erroneously double request a transaction we just confirmed.